bot.py

```diff
@@ -0,0 +1,77 @@
+import heuristics
+import board_heu
+from game import Game, Player, BOARD_SIZE
+from draughts import Board, Move
+import time
+
+"""
+NEW alpha-beta pruning
+
+TO DO:
+- Clean up code
+- For now, everything minimax-related is played on the Board (not Game) 
+  --> change that
+- Problem with moves that are evaluated the same: the model will have a tendency to play a little too much on the left (or right) 
+  --> implement some randomness
+- Minimax takes too long to compute 
+  --> alpha-beta pruning
+- Update alpha-beta pruning in all the files
+
+Minimax handles jump chains as one action.
+"""
+
+def minimax(board: Board, depth: int, alpha: int, beta: int, max_player: bool):
+    """
+    Minimax algorithm
+    Returns the score, the move and the resulting board of the best move to play according to heuristics
+    """
+    if depth == 0 or board.is_over():
+        # return board_heu.piece_count(board), board.move_stack[-1].steps_move, board
+        return board_heu.piece_count(board), board.move_stack[-1]
+    if max_player:
+        max_eval = float('-inf')
+        best_board = None
+        for child in get_children(board):
```

Comment:
To make this work with Game instead of board, we could replace this:
```python
for child in get_children(board):
    evaluation = minimax(child, ...)
```

with this:
```python
for action in game.get_legal_moves():
    game.play(action)
    evaluation = minimax(game, ...)
    game.undo(action)
```
And make it so that minimax works with the game directly instead of the board.

Reply:
Another advantage is that we don't need to store best_board and access the move_stack but we can just store the best_action and return this directly